src/main/java/object/manager/GameObjectManager.java

```diff
@@ -0,0 +1,63 @@
+package object.manager;
+
+import object.Bullet;
+import object.GameObject;
+import object.Tank;
+
+public class GameObjectManager {
+
+    private final PhysicObjectManager physicObjectManager;
+
+    private final TankObjectManager tankObjectManager;
+    private final BulletObjectManager bulletObjectManager;
+    private final BonusObjectManager bonusObjectManager;
+
+    public GameObjectManager() {
+        this.physicObjectManager = new PhysicObjectManager();
+        this.tankObjectManager = new TankObjectManager();
+        this.bulletObjectManager = new BulletObjectManager();
+        this.bonusObjectManager = new BonusObjectManager();
+    }
+
+    public PhysicObjectManager getPhysicObjectManager() {
+        return physicObjectManager;
+    }
+
+    public void removeObject(GameObject gameObject) {
+        physicObjectManager.removeObject(gameObject);
+
+        if (tankObjectManager.getTypeId() == gameObject.getTypeId()) {
+            tankObjectManager.removeById(gameObject.getId());
+        }
+
+        if (bulletObjectManager.getTypeId() == gameObject.getTypeId()) {
+            bulletObjectManager.removeById(gameObject.getId());
+        }
+
+        if (bonusObjectManager.getTypeId() == gameObject.getTypeId()) {
+            bonusObjectManager.removeById(gameObject.getId());
+        }
+    }
+
+    public void addTank(Tank tank) {
+        tankObjectManager.addObject(tank);
+        physicObjectManager.addObject(tank);
```

Comment:
code duplication

Reply:
it's not a duplication, `PhysicObjectManager` should know somehow about objects from others `ObjectManagers`

Reply:
@ramazan-bagaev I've ment that physicObjectManager.addObject(tank); is called twice from addTank() and addBullet() but should have been extracted to separate private method